### PR TITLE
Adding MenAnomaly and True Anomaly display and dialog

### PR DESCRIPTION
**Pull request by NoFoolLikeOne**
https://github.com/user-attachments/assets/005b71a9-b005-4e0b-a69f-3e5fc75fe728

image

### CODE REVIEW

File: src/app/dialogs/anomaly-dialog/anomaly-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+          M = {{ meanAnomalyLive() | number:'0.1-1' }}°
+          @if (hasEccentricity) { , ν = {{ trueAnomalyLive() | number:'0.1-1' }}° }
+        </li>
+      </ul>
+    </div>
+
+    @if (diagram(); as d) {
+    <div class="diagram">
+      <svg viewBox="0 0 120 120" class="anomaly-diagram" role="img" aria-label="Mean vs. true anomaly diagram">
+        <!-- Periapsis reference direction -->
```
Comment (DarkSession) [inline src/app/dialogs/anomaly-dialog/anomaly-dialog.component.html:25]: Nit: this comment is misleading. `referenceLine` is the fixed **0° reference direction** (ascending node), not the periapsis direction — it always runs along +x from the focus regardless of `argOfPeriapsis`, matching the `AnomalyDiagram` interface doc in `orbital-diagrams.ts`. When `arg ≠ 0` the periapsis points elsewhere. Suggest "0° reference direction" for consistency. Non-blocking.

File: src/app/data/orbital-diagrams.ts [hunk 2]
```diff
@@ -238,6 +238,72 @@ export function periapsisDiagram(argDeg: number, eccentricity?: number): Periaps
...
+ * their real orbits, making their true-anomaly markers land in unrelated places even
+ * though the bodies are at the same physical position.
+ */
+export function anomalyDiagram(meanAnomalyDeg: number, trueAnomalyDeg: number, eccentricity?: number, argOfPeriapsisDeg?: number): AnomalyDiagram {
+  const M = Number.isFinite(meanAnomalyDeg) ? meanAnomalyDeg : 0;
+  const nu = Number.isFinite(trueAnomalyDeg) ? trueAnomalyDeg : 0;
+  const arg = Number.isFinite(argOfPeriapsisDeg as number) ? (argOfPeriapsisDeg as number) : 0;
+  // Unlike periapsisDiagram, e = 0 is a valid and useful case here: a circular orbit is
+  // exactly where M and ν coincide, which is part of what this diagram teaches.
+  const e = clamp(Number.isFinite(eccentricity as number) ? (eccentricity as number) : 0.5, 0, 0.85);
```
Comment (DarkSession) [inline src/app/data/orbital-diagrams.ts:273]: Minor (cosmetic): `e` is clamped to `[0, 0.85]` here, but the `ν` passed in is derived by `meanToTrueAnomaly` with `e` clamped to `0.999`. For orbits with `e ∈ (0.85, 0.999]` the true-anomaly marker's angle uses the real eccentricity while its radius uses the clamped one, so the marker won't sit exactly on the drawn ellipse. Affects only very eccentric orbits (rare in ED) and is an acceptable trade-off for drawability — flagging so it's a conscious choice. Non-blocking.

**Review by DarkSession — APPROVED** [review]
✅ **Reviewed — looks good to merge.**

The physics is correct (verified `anomalyDiagram` against Kepler's equation: markers coincide at periapsis/apoapsis and for e=0, and true anomaly correctly leads mean in the first half-orbit). The code closely mirrors the existing `periapsisDiagram` / `ApoPeriDialogComponent` patterns, all 636 unit tests pass, and `ng build` (strict template typecheck) succeeds with the dialog correctly lazy-loaded. Test coverage is thorough — geometry edge cases, epoch tree-walking, and missing-eccentricity / missing-telemetry fallbacks.

A couple of minor, non-blocking notes inline. No correctness issues found.

**Review by DarkSession — APPROVED** [review]
